app/models/profile.rb

```diff
@@ -97,7 +101,7 @@ def activate_after_password_reset
       update!(
         deactivation_reason: nil,
       )
-      activate
+      activate(reason_deactivated: :password_reset)
```

Comment:
I like this pattern! Then we can coalesce the behavior for different reasons inside activate.